### ISSUE TITLE
make behave fails on BSD for fresh clone of gpdb

### ISSUE THREAD

**Issue by kalensk** [greenplum-db/gpdb#7237]
### Greenplum version or build
master (GPDB 7)

PostgreSQL 9.4.20 (Greenplum Database 6.0.0-alpha.0+dev.16122.geb0647183f build dev) on x86_64-apple-darwin17.5.0, compiled by Apple LLVM version 9.1.0 (clang-902.0.39.2), 64-bit compiled on Mar 21 2019 16:31:20 (with assert checking)

### OS version and uname -a

Mac High Sierra 10.13.4

Darwin kalen 17.5.0 Darwin Kernel Version 17.5.0: Mon Mar  5 22:24:32 PST 2018; root:xnu-4570.51.1~1/RELEASE_X86_64 x86_64

### autoconf options used ( config.status --config )

 $ ./configure --prefix=/usr/local/gpdb7 --disable-orca --enable-depend --enable-debug --enable-cassert --disable-pxf --with-perl --with-python --with-libxml --with-gssapi CFLAGS=-O0 CXXFLAGS=-O0

### Installation information ( pg_config )

BINDIR = /usr/local/gpdb7/bin
DOCDIR = /usr/local/gpdb7/share/doc/postgresql
HTMLDIR = /usr/local/gpdb7/share/doc/postgresql
INCLUDEDIR = /usr/local/gpdb7/include
PKGINCLUDEDIR = /usr/local/gpdb7/include/postgresql
INCLUDEDIR-SERVER = /usr/local/gpdb7/include/postgresql/server
LIBDIR = /usr/local/gpdb7/lib
PKGLIBDIR = /usr/local/gpdb7/lib/postgresql
LOCALEDIR = /usr/local/gpdb7/share/locale
MANDIR = /usr/local/gpdb7/share/man
SHAREDIR = /usr/local/gpdb7/share/postgresql
SYSCONFDIR = /usr/local/gpdb7/etc/postgresql
PGXS = /usr/local/gpdb7/lib/postgresql/pgxs/src/makefiles/pgxs.mk
CONFIGURE = '--prefix=/usr/local/gpdb7' '--disable-orca' '--enable-depend' '--enable-debug' '--enable-cassert' '--disable-pxf' '--with-perl' '--with-python' '--with-libxml' '--with-gssapi' 'CFLAGS=-O0' 'CXXFLAGS=-O0'
CC = gcc
CPPFLAGS = -I/Library/Developer/CommandLineTools/SDKs/MacOSX10.13.sdk/usr/include/libxml2 -I/usr/local/gpdb7/include
CFLAGS = -Wall -Wmissing-prototypes -Wpointer-arith -Wendif-labels -Wmissing-format-attribute -Wformat-security -fno-strict-aliasing -fwrapv -Wno-address -Wno-unused-command-line-argument -g -ggdb -O0 -std=gnu99 -Werror=uninitialized -Werror=implicit-function-declaration -I/usr/local/gpdb7/include
CFLAGS_SL =
LDFLAGS = -L/Library/Developer/CommandLineTools/SDKs/MacOSX10.13.sdk/usr/lib -Wl,-dead_strip_dylibs -L/usr/local/gpdb7/lib
LDFLAGS_EX =
LDFLAGS_SL =
LIBS = -lpgcommon -lpgport -lxml2 -lgssapi_krb5 -lzstd -lm -L/usr/local/gpdb7/lib
VERSION = PostgreSQL 9.4.20

### Expected behavior

make behave is successful on BSD for a fresh clone of GPDB

### Actual behavior

cd gpMgmt &&  make behave

```
sed: 1: "setup.py": unterminated substitute pattern
make[1]: *** [/Users/kkrempely/workspace/gpdb7/gpMgmt/bin/pythonSrc/ext/install/bin/behave] Error 1
make: *** [/Users/kkrempely/workspace/gpdb7/gpMgmt/bin/pythonSrc/ext/install/bin/behave] Error 2
```

Fix was adding `-e` to the sed command on Mac in the [gpMgmt/bin/Makefile](https://github.com/greenplum-db/gpdb/blob/d0839b23dad8d46276c1724d0f21aebb7aa600f8/gpMgmt/bin/Makefile#L243). `sed -i -e 's/parse_type>=0.3.4/parse_type==0.4.1/' setup.py \`

Basically the difference between BSD and GNU sed for `-i` extension files.

Some thoughts:
- Do we want to make gsed a prerequisite for running on BSD, and in our makefile detect if the user is running Darwin then set SED_BIN=$(gsed), etc.?
- Do we want to generate the backup file from -i and then delete it when on Darwin, as outlined [here](https://stackoverflow.com/questions/5694228/sed-in-place-flag-that-works-both-on-mac-bsd-and-linux).
- Do we want to simply raise a more obvious error message? 
- ,,,?
